### PR TITLE
Added ST_TILEENVELOPE to "constructors" module

### PR DESCRIPTION
**Pull request by alex-graciano**
The original PostGIS function (https://postgis.net/docs/ST_TileEnvelope.html) also includes an extra `margin` parameter to grow a tile by the given percentage. As we stated in Slack, the current implementation uses just the (x,y,z) index to extract the polygon but in case the `margin`parameter is further required, this implementation is not valid and we must go with a previous step for extracting the bounding box. We can use something like this:

```sql
    SELECT `@@BQ_PROJECTID@@.@@BQ_DATASET_CONSTRUCTORS@@.ST_MAKEENVELOPE`(
        bounds[OFFSET(0)],
        bounds[OFFSET(1)],
        bounds[OFFSET(2)],
        bounds[OFFSET(3)]
    )
    FROM (
        SELECT `@@BQ_PROJECTID@@.@@BQ_DATASET_QUADKEY@@.BBOX`(
            `@@BQ_PROJECTID@@.@@BQ_DATASET_QUADKEY@@.QUADINT_FROMZXY`(zoomLevel,xTile,yTile)) AS bounds
    )
```

The problem is the `ST_MAKEPOLYGON` used in our `ST_MAKEENVELOPE` has some issues with dealing with lower zoom levels, splitting the resulting polygon in a weird fashion, e.g. giving this result for the (0,0,0) tile:

`GEOMETRYCOLLECTION(POLYGON((0 0, 120 0, -120 0, 0 0)), POLYGON((0 0, -120 0, 120 0, 0 0)))`

In conclusion, I would keep the signature as it is, just tanking the tile index.

### CODE REVIEW

**Review by Jesus89 — APPROVED** [review]
LGTM

**Comment by Jesus89** [comment]
Tested in staging :+1:

**Review by vdelacruzb — APPROVED** [review]
Nice
